src/core/meshcop/dataset_manager.cpp

```diff
@@ -337,16 +337,20 @@ void DatasetManager::HandleCoapResponse(void *               aContext,
                                         const otMessageInfo *aMessageInfo,
                                         Error                aError)
 {
-    OT_UNUSED_VARIABLE(aMessage);
-    OT_UNUSED_VARIABLE(aMessageInfo);
-    OT_UNUSED_VARIABLE(aError);
-
-    static_cast<DatasetManager *>(aContext)->HandleCoapResponse();
+    static_cast<DatasetManager *>(aContext)->HandleCoapResponse(
+        *static_cast<Coap::Message *>(aMessage), *static_cast<const Ip6::MessageInfo *>(aMessageInfo), aError);
 }
 
-void DatasetManager::HandleCoapResponse(void)
+void DatasetManager::HandleCoapResponse(Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo, Error aError)
 {
-    mCoapPending = false;
+    OT_UNUSED_VARIABLE(aMessageInfo);
+
+    if (mMgmtSetCallback != nullptr)
+    {
+        CallMgmtSetCallback(aMessage, aError);
+    }
+
+    mMgmtPending = false;
     mTimer.Start(kSendSetDelay);
```

Comment:
might enhance this and not trigger the timer for Mgmt API call?

Reply:
Done.